src/box_utils.cpp

```diff
@@ -0,0 +1,170 @@
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ *  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <algorithm>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <geometry_msgs/Point32.h>
+#include <ros/ros.h>
+
+#include "box_utils.h"
+
+std::string boxToString(const Box& box)
+{
+  std::stringstream box_stringstream;
+  box_stringstream << "["
+                   << "[" << box.min.x << ", " << box.min.y << ", " << box.min.z << "]"
+                   << ", "
+                   << "[" << box.max.x << ", " << box.max.y << ", " << box.max.z + "]"
+                   << "]";
+
+  return box_stringstream.str();
+}
+
+Box makeBoxFromTwoPoints(const geometry_msgs::Point32& point0, const geometry_msgs::Point32& point1)
+{
+  Box box;
+
+  box.min.x = std::min(point0.x, point1.x);
+  box.min.y = std::min(point0.y, point1.y);
+  box.min.z = std::min(point0.z, point1.z);
+
+  box.max.x = std::max(point0.x, point1.x);
+  box.max.y = std::max(point0.y, point1.y);
+  box.max.z = std::max(point0.z, point1.z);
+
+  return box;
+}
+
+Box makeBoxFromXMLRPC(const XmlRpc::XmlRpcValue& box_xmlrpc, const std::string& full_param_name)
+{
+  // checks if an array has just two elements.
+  if (box_xmlrpc.getType() != XmlRpc::XmlRpcValue::TypeArray || box_xmlrpc.size() != 2)
+  {
+    ROS_FATAL("The box (parameter %s) must be specified as nested list on the parameter server with two points"
+              "i.e. [[x1, y1, z1], [x2, y2, z2]]",
+              full_param_name.c_str());
+
+    throw std::runtime_error("The box must be specified as nested list on the parameter server with two points"
+                             "i.e. [[x1, y1, z1], [x2, y2, z2]]");
+  }
+
+  std::vector<geometry_msgs::Point32> points;
+  points.resize(box_xmlrpc.size());  // points.size() == 2
+
+  for (int i = 0; i < box_xmlrpc.size(); ++i)
+  {
+    // checks if each element (point) has three elements,
+    // i.e. x, y, and z coordinates
+    XmlRpc::XmlRpcValue point_xmlrpc = box_xmlrpc[i];
+    if (point_xmlrpc.getType() != XmlRpc::XmlRpcValue::TypeArray || point_xmlrpc.size() != 3)
+    {
+      ROS_FATAL("The box (parameter %s) must be specified as list of lists,"
+                "i.e. [[x1, y1, z1], [x2, y2, z2]], but this spec is not of that form.",
+                full_param_name.c_str());
+      throw std::runtime_error("The box must be specified as list of lists,"
+                               "i.e. [[x1, y1, z1], [x2, y2, z2]], but this spec is not of that form");
+    }
+
+    points[i].x = getNumberFromXMLRPC(point_xmlrpc[0], full_param_name);
+    points[i].y = getNumberFromXMLRPC(point_xmlrpc[1], full_param_name);
+    points[i].z = getNumberFromXMLRPC(point_xmlrpc[2], full_param_name);
+  }
+
+  Box box = makeBoxFromTwoPoints(points[0], points[1]);
+  return box;
```

Comment:
I think removing the `box` is better to reduce code.

```suggestion
  return makeBoxFromTwoPoints(points[0], points[1]);
```